### DISCUSSION TITLE
How to properly cleanup a longhorn-system namespace in "Terminating" state, with webhook errors when trying to remove the resources?

### DISCUSSION THREAD

**Discussion by johanneskastl** [longhorn/longhorn#9167] [Q&A]
Hi all,

somehow ArgoCD messed up my install of Longhorn on Talos (do not ask me how, I still need to understand what went wrong), and now I am left with a longhorn-system namespace stuck in `Terminating` state.

There are several `setting.longhorn.io` resources in this namespace (which block its deletion, I guess).

- I cannot delete any resources in it, due to webhook errors (there is nothing receiving the webhook anymore)
- I cannot use the "Uninstall with kubectl" method, as I cannot create the serviceAccount and the job as the namespace is in `Terminating` state

I found some similar issues/discussions, but no solution that would work for me...

- https://github.com/longhorn/longhorn/discussions/5318
- https://github.com/longhorn/longhorn/issues/5319

So, how to properly clean up this mess? I would rather not destroy the whole cluster and start from scratch, but if I find no solution soon I might have to do exactly that...

Kind Regards,
Johannes

**Comment by johanneskastl** [comment]
> So, how to properly clean up this mess? I would rather not destroy the whole cluster and start from scratch, but if I find no solution soon I might have to do exactly that...

To not be blocked by this issue I decided to destroy and clean up the cluster and create a new cluster from scratch. Automation and GitOps for the win...

But I would still be interested in a way to properly clean up a namespace in "Terminating" state with resources that cannot be deleted due to missing webhooks...

**Reply by thiduzz** [reply]
1. Cleanup all finalizers for the CRDs:
```
for crd in $(kubectl get crd -o name | grep longhorn); do
  kubectl patch $crd -p '{"metadata":{"finalizers":[]}}' --type=merge
done
```
2. Add the deleting-confirmation flag:
```
kubectl -n longhorn-system patch -p '{"value": "true"}' --type=merge lhs deleting-confirmation-flag
```
3. Effectively delete the CRDs:
```
kubectl delete crd -l longhorn-manager
```
4. Delete the namespace:
```
kubectl delete namespace longhorn-system --grace-period=0 --force
```
5. Wait some minutes (I didnt track time - maybe hours)

**Reply by QuinnBast** [reply]
```
kubectl -n longhorn-system patch -p '{"value": "true"}' --type=merge lhs deleting-confirmation-flag
```

Doesn't work since the namespace is being torn down. Says that the admission webhook is not found :/

Also hours to delete is insane.
